pkg/internal/diskutil.go

```diff
@@ -0,0 +1,223 @@
+}
+
+func parseBitBool(s string) (bool, error) {
+	if s == "0" || s == "" {
+		return false, nil
+	} else if s == "1" {
+		return true, nil
+	}
+	return false, fmt.Errorf("lsblk bool value not 0 or 1: %q", s)
+}
+
+// GetSize as int64
+func (b BlockDevice) GetSize() (int64, error) {
+	v, err := strconv.ParseInt(b.Size, 10, 64)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to parse size property %q as int64", b.Size)
+	}
+	return v, err
+}
+
+// HasChildren check on BlockDevice
+func (b BlockDevice) HasChildren() (bool, error) {
+	sysDevDir := filepath.Join("/sys/block/", b.Name, "/*")
+	paths, err := filepath.Glob(sysDevDir)
+	if err != nil {
+		return false, errors.Wrapf(err, "could not glob path: %q to verify partiions", sysDevDir)
+	}
+	for _, path := range paths {
+		name := filepath.Base(path)
+		if strings.HasPrefix(name, b.Name) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// GetPathByID check on BlockDevice
+func (b BlockDevice) GetPathByID() (string, error) {
+
+	// return if previously populated value is valid
+	if len(b.pathByID) > 0 && strings.HasPrefix(b.pathByID, DiskByIDDir) {
+		evalsCorrectly, err := PathEvalsToDiskLabel(b.pathByID, b.Name)
+		if err == nil && evalsCorrectly {
+			return b.pathByID, nil
+		}
+	}
+	b.pathByID = ""
+	diskByIDDir := filepath.Join(DiskByIDDir, "/*")
+	paths, err := filepath.Glob(diskByIDDir)
+	if err != nil {
+		return "", fmt.Errorf("could not list files in %q: %w", DiskByIDDir, err)
+	}
+	for _, path := range paths {
+		isMatch, err := PathEvalsToDiskLabel(path, b.Name)
+		if err != nil {
+			return "", err
+		}
+		if isMatch {
+			b.pathByID = path
+			return path, nil
+		}
+	}
+	// return path by label and error
+	return fmt.Sprintf("/dev/%s", b.Name), IDPathNotFoundError{DeviceName: b.Name}
+}
+
+// PathEvalsToDiskLabel checks if the path is a symplink to a file devName
+func PathEvalsToDiskLabel(path, devName string) (bool, error) {
+	devPath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return false, errors.Wrapf(err, "could not eval symLink %q", path)
+	}
+	if filepath.Base(devPath) == devName {
+		return true, nil
+	}
+	return false, nil
+}
+
+// ListBlockDevices using the lsblk command
+func ListBlockDevices() ([]BlockDevice, []string, error) {
+	var output bytes.Buffer
+	var blockDevices []BlockDevice
+
+	columns := "NAME,ROTA,TYPE,SIZE,MODEL,VENDOR,RO,RM,STATE,FSTYPE"
+	args := []string{"--pairs", "-b", "-o", columns}
+	cmd := exec.Command("lsblk", args...)
+	cmd.Stdout = &output
+	err := cmd.Run()
+	if err != nil {
+		return []BlockDevice{}, []string{}, err
+	}
+
+	badRows := make([]string, 0)
+	// convert to json and then Marshal.
+	outputMapList := make([]map[string]interface{}, 0)
+	rowList := strings.Split(output.String(), "\n")
+RowLoop:
+	for _, row := range rowList {
+		if len(strings.Trim(row, " ")) == 0 {
+			break RowLoop
+		}
+		outputMap := make(map[string]interface{})
+		// split by `" ` to avoid splitting on spaces in MODEL,VENDOR
```

Comment:
Can we also trim the white spaces in model and vendor names.

Reply:
I think we should include whatever is between the quoutes in the output. Either way, the matcher only depends on whether the included value is a substring of this value.